File: chaveamento-robos/src/main/java/org/chaveamento/model/partida/Partida.java
package org.chaveamento.model.partida;

import jakarta.persistence.*;
import lombok.*;
import org.chaveamento.model.torneio.Torneio;
import org.chaveamento.model.time.Time;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "partidas")
public class Partida {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private FasePartida fase;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private StatusPartida status;

    @ManyToOne
    @JoinColumn(name = "time_a_id")
    private Time timeA;

    @ManyToOne
    @JoinColumn(name = "time_b_id")
    private Time timeB;

    private Integer placarTimeA;

    private Integer placarTimeB;

    @ManyToOne
    @JoinColumn(name = "vencedor_id")
    private Time vencedor;

    @ManyToOne
    @JoinColumn(name = "proxima_partida_id")
    private Partida proximaPartida;

    @ManyToOne
    @JoinColumn(name = "torneio_id", nullable = false)
    private Torneio torneio;

    @Enumerated(EnumType.STRING)
    private PosicaoProximaPartida posicaoProximaPartida;
}

File: front-chaveamento-robos/src/types/partida.ts
export type FasePartida = "OITAVAS" | "QUARTAS" | "SEMIFINAL" | "FINAL";

export type StatusPartida = "AGUARDANDO" | "EM_ANDAMENTO" | "FINALIZADA";

export interface Partida {
  id: number;
  fase: FasePartida;
  status: StatusPartida;
  torneioId: number;
  timeAId: number | null;
  timeBId: number | null;
  placarTimeA: number | null;
  placarTimeB: number | null;
  vencedorId: number | null;
  proximaPartidaId: number | null;
}

export interface ResultadoPartidaRequest {
  placarTimeA: number;
  placarTimeB: number;
}
